editor/components/multi-select-scroll-into-view/index.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * External dependencies
+ */
+import scrollIntoView from 'dom-scroll-into-view';
+
+/**
+ * WordPress dependencies
+ */
+import { Component } from '@wordpress/element';
+import { query } from '@wordpress/data';
+
+class MultiSelectScrollIntoView extends Component {
+	componentDidUpdate() {
+		// Relies on expectation that `componentDidUpdate` will only be called
+		// if value of `extentUID` changes.
+		this.scrollIntoView();
+	}
+
+	/**
+	 * Ensures that if a multi-selection exists, the extent of the selection is
+	 * visible within the viewport.
+	 *
+	 * @return {void}
+	 */
+	scrollIntoView() {
+		const { extentUID } = this.props;
+		if ( ! extentUID ) {
+			return;
+		}
+
+		const extentNode = document.querySelector( '[data-block="' + extentUID + '"]' );
+		if ( ! extentNode ) {
+			return;
+		}
+
+		scrollIntoView(
+			extentNode,
+			extentNode.closest( '.edit-post-layout__content' ),
+			{ onlyScrollIfNeeded: true }
+		);
+	}
+
+	render() {
+		return null;
+	}
+}
+
+export default query( ( select ) => {
```

Comment:
Any reason for using query here instead of connect

Reply:
>Any reason for using query here instead of connect

I guess it speaks to a broader question: Do we want to start discouraging use of `connect` altogether? My thinking is that it speaks strongly to its viability if we're forced to use it in first-party code. Also a nice abstraction to avoid interacting with Redux directly. Also avoids some problems with variable shadowing and importing selectors.

Reply:
> Also avoids some problems with variable shadowing and importing selectors.

Can you elaborate a little? Or are these separate things, i.e. you mean variable shadowing for actions (see our [recent discussion](https://github.com/WordPress/gutenberg/pull/4839#issuecomment-363088280))?

Reply:
>Can you elaborate a little?

I'll elaborate in the form of a refactoring pull request demonstrating the benefits 😆 

See: https://github.com/WordPress/gutenberg/pull/5083